backend/inspirehep/records/marshmallow/literature/latex.py

```diff
@@ -113,3 +122,29 @@ def get_dois(self, data):
         for doi_data in dois:
             doi_data["value"] = latex_encode(doi_data.get("value"))
         return dois or None
+
+    def get_book_publication_info(self, data):
+        if "book" in get_value(data, "document_type", []):
+            imprint_date = get_value(data, "imprints.date[0]")
+            publisher = latex_encode(get_value(data, "imprints.publisher[0]"))
+            if imprint_date and publisher:
+                imprint_year = imprint_date.split("-")[0]
+                return f"{publisher}, {imprint_year}"
+            elif imprint_date:
+                return imprint_date.split("-")[0]
+            elif publisher:
+                return publisher
+        return missing
+
+    def get_isbn(self, data):
+        def hyphenate_if_possible(no_hyphens):
+            try:
+                return normalize_isbn(no_hyphens)
+            except ISBNError:
+                return no_hyphens
+
+        if "book" in get_value(data, "document_type", []):
```

Comment:
again, I would return here if it's not in `document_type`. Also, if I am correct only books have `isbns` - if that's the case you can get rid of this line and just check `if isbns`